connectors/vmware/src/centreon/script/centreonvault.pm

```diff
@@ -59,52 +106,24 @@ sub check_configuration {
         $self->{logger}->writeLogInfo("Vault port is missing from configuration.");
         $self->{vault_config}->{port} = '443';
     }
-}
-
-sub init {
-    my ($self, %options) = @_;
 
-    $self->check_options();
-
-    # Since this class is used, we assume that credentials are encrypted. It may be reconsidered later.
-    $self->{credentials_are_encrypted} = 1;
-
-    # check if the following information is available
-    $self->{logger}->writeLogDebug("Reading Vault configuration from file " . $self->{config_file} . ".");
-    $self->{vault_config} = centreon::vmware::common::parse_json_file( 'json_file' => $self->{config_file} ) or return 0;
-
-    $self->check_configuration();
-
-    $self->{logger}->writeLogDebug("Vault configuration read. Name: " . $self->{vault_config}->{name} . ". Url: " . $self->{vault_config}->{url} . ".");
-
-    # Normally, the role-id and secret-id data are encrypted using AES wit the following information:
+    # Normally, the role_id and secret_id data are encrypted using AES wit the following information:
     # firstKey = APP_SECRET (environment variable)
-# secondKey = 'salt' key given by vault.json configuration file
+    # secondKey = 'salt' (hashing) key given by vault.json configuration file
     # both are base64 encoded
     if ( !defined($self->{vault_config}->{salt}) || $self->{vault_config}->{salt} eq '') {
-        $self->{logger}->writeLogError("Vault environment does not seem complete: 'salt' attribute missing from " . $self->{config_file} . ". 'role-id' and 'secret-id' won't be decrypted, so they'll be uses as they're stored in the vault config file.");
-        $self->{credentials_are_encrypted} = 0;
+        $self->{logger}->writeLogError("Vault environment does not seem complete: 'salt' attribute missing from "
+            . $self->{config_file}
+            . ". 'role_id' and 'secret_id' won't be decrypted, so they'll be used as they're stored in the vault config file.");
```

Comment:
the behaviour right now after returning undef is to never use the vault.
Either change the behaviour of the error message.